linux-installer.sh

```diff
@@ -91,44 +109,85 @@ has_wget() {
 # Stepping:            9
 # CPU MHz:             2900.000
 has_lscpu() {
-  echo "TODO"
+   if [[ -n "$(command -v lscpu)" ]]; then
+      return 0
+   fi
+   return 1
 }
 
 # Check if we have /proc/cpuinfo
 has_proc_cpuinfo() {
-  echo "TODO"
+   if [[ -e /proc/cpuinfo ]]; then
+      return 0
+   fi
+   return 1
+}
+
+check_ark_intel() {
+   productUrl=$(curl "https://ark.intel.com/search/AutoComplete?term=${cpuModel}" | sed -n 's/.*\"quickUrl\":\"\(.*\)\".*/\1/p')
+   tmpCpuType=$(curl --silent "https://ark.intel.com${productUrl}" | sed -n 's/.*Products formerly \(.*\)<.*/\1/p')
+   CPU_TYPE=$(echo $tmpCpuType | sed 's/ //g' | awk '{print tolower($0)}')
 }
 
 # Check CPU type
 check_cpu_type() {
-  echo "TODO"
-  # If we have lscpu, use the following command:
-
-  # lscpu | sed -nr '/Model name/ s/.*:\s*(.*) @ .*/\1/p' | cut -d ' ' -f 3
-  # Example: i7-7820HQ
-
-  # If we have /proc/cpuinfo, use the following command:
-  # cat /proc/cpuinfo | sed -nr '/model name/ s/.*:\s*(.*) @ .*/\1/p' | cut -d ' ' -f 3 | head -1
-  # Example: i7-7820HQ
+   CPU_TYPE="core2"
+
+   if has_lscpu; then
+      cpuModel=$(lscpu | sed -nr '/Model name/ s/.*:\s*(.*) @ .*/\1/p' | cut -d ' ' -f 3)
+      check_ark_intel
+   elif has_proc_cpuinfo; then
+      cpuModel=$(cat /proc/cpuinfo | sed -nr '/model name/ s/.*:\s*(.*) @ .*/\1/p' | cut -d ' ' -f 3 | head -1)
+      check_ark_intel
+   fi
+}
 
-  # If we get something, look it up against our lookup tables
-  # Otherwise, use the `compatible` version (ie. core2)
-  CPU_TYPE="core2"
+install_curl() {
+   if has_yum; then
+      sudo yum upgrade -y
+      sudo yum install curl
+   elif has_apt; then
+      sudo apt-get update -y
+      sudo apt-get install curl
+   elif has_apk; then
+      sudo apk add curl
+   fi
 }
 
 # Script starts here!
 if [[ -z "$WALLET_ADDRESS" ]]; then
   echo "WALLET_ADDRESS was not defined!"
   exit 1
 fi
-if [[ -z ${WORKER_ID+x} ]]; then
+if [[ -z ${WORKER_ID} ]]; then
```

Comment:
LOL 😂 
The `+x` was meant to be there; the thing is, without the `+x`, there's no way to tell the difference between an empty string, (ie. `WORKER_ID=""`) and unset.